pkg/metrics/metrics.go

```diff
@@ -0,0 +1,185 @@
+	mc.collectMachineSetMetrics(ch)
+}
+
+// Describe implements the prometheus.Collector interface.
+func (mc MachineCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- MachineCountDesc
+	ch <- MachineSetCountDesc
+}
+
+// Collect implements the prometheus.Collector interface.
+func (mc MachineCollector) collectMachineMetrics(ch chan<- prometheus.Metric) {
+	machineList, err := mc.listMachines()
+	if err != nil {
+		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machine-count", "reason": err.Error()}).Inc()
+		return
+	}
+
+	for _, machine := range machineList {
+		mMeta := machine.ObjectMeta
+		typeMeta := machine.TypeMeta
+		mSpec := machine.Spec
+		providerid := ""
+		if mSpec.ProviderID != nil {
+			providerid = *mSpec.ProviderID
+		}
+
+		MachineInfo.With(prometheus.Labels{
+			"name":             mMeta.Name,
+			"namespace":        mMeta.Namespace,
+			"createdAt":        strconv.FormatInt(mMeta.GetCreationTimestamp().Time.Unix(), 10),
+			"spec_provider_id": providerid,
+			"api_version":      typeMeta.APIVersion,
+			"node":             machine.Status.NodeRef.Name}).Set(float64(1))
+	}
+
+	metric, err := prometheus.NewConstMetric(MachineCountDesc, prometheus.GaugeValue, float64(len(machineList)))
+	if err != nil {
+		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machine-count", "reason": err.Error()}).Inc()
+		return
+	}
+	ch <- metric
+	glog.V(4).Infof("collectmachineMetrics exit")
+
+}
+
+// collectMachineSetMetrics is method to collect machineSet related metrics.
+func (mc MachineCollector) collectMachineSetMetrics(ch chan<- prometheus.Metric) {
+	machineSetList, err := mc.listMachineSets()
+	if err != nil {
+		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machineset-count", "reason": err.Error()}).Inc()
+		return
+	}
+	metric, err := prometheus.NewConstMetric(MachineSetCountDesc, prometheus.GaugeValue, float64(len(machineSetList)))
+	if err != nil {
+		ScrapeFailedCounter.With(prometheus.Labels{"kind": "Machineset-count", "reason": err.Error()}).Inc()
+		return
+	}
+	ch <- metric
+
+	for _, machineSet := range machineSetList {
+
+		MachineSetInfo.With(prometheus.Labels{
+			"name":        machineSet.Name,
+			"namespace":   machineSet.Namespace,
+			"createdAt":   strconv.FormatInt(machineSet.GetCreationTimestamp().Time.Unix(), 10),
+			"api_version": machineSet.TypeMeta.APIVersion}).Set(float64(1))
+
+		MachineSetInfoSpecReplicas.With(prometheus.Labels{
+			"name":      machineSet.Name,
+			"namespace": machineSet.Namespace}).Set(float64(pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0)))
+		MachineSetStatusAvailableReplicas.With(prometheus.Labels{
+			"name":      machineSet.Name,
+			"namespace": machineSet.Namespace,
+		}).Set(float64(machineSet.Status.AvailableReplicas))
+
+		MachineSetStatusReadyReplicas.With(prometheus.Labels{
+			"name":      machineSet.Name,
+			"namespace": machineSet.Namespace,
+		}).Set(float64(machineSet.Status.ReadyReplicas))
+
+		MachineSetStatusReplicas.With(prometheus.Labels{
```

Comment:
what is this, can we drop it? seems a copy of the above `MachineSetStatusReadyReplicas`

Reply:
This is for Status.Replicas reporting. Fixed the typo error